v-next/hardhat/src/internal/helpers/config-loading.ts

```diff
@@ -0,0 +1,61 @@
+import { isAbsolute, resolve } from "node:path";
+import { pathToFileURL } from "node:url";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+import { findUp } from "@nomicfoundation/hardhat-utils/fs";
+
+import { ERRORS } from "../../../../hardhat-errors/src/descriptors.js";
+
+async function findClosestHardhatConfig(): Promise<string> {
+  let hardhatConfigPath = await findUp("hardhat.config.ts");
+
+  if (hardhatConfigPath !== undefined) {
+    return hardhatConfigPath;
+  }
+
+  hardhatConfigPath = await findUp("hardhat.config.js");
+
+  if (hardhatConfigPath !== undefined) {
+    return hardhatConfigPath;
+  }
+
+  throw new HardhatError(ERRORS.GENERAL.NO_CONFIG_FILE_FOUND);
+}
+
+export async function resolveConfigPath(): Promise<string> {
+  const configPath = process.env.HARDHAT_CONFIG;
+
+  if (configPath !== undefined) {
+    return configPath;
+  }
+
+  return findClosestHardhatConfig();
+}
+
+export async function importUserConfig(configPath: string) {
+  const normalizedPath = isAbsolute(configPath)
+    ? configPath
+    : resolve(process.cwd(), configPath);
+
+  const { exists } = await import("@nomicfoundation/hardhat-utils/fs");
```

Comment:
We can probably use regular imports for the `hardhat-utils` package, as the package only imports things from node and should be fast enough to avoid using dynamic imports.

Reply:
Ignore this comment for now, we'll need more data to decide if and when it is worth using dynamic imports, as most things now will be lazy and wont benefit from using this the pattern.

Reply:
This code in particular already existed in the `v-next` branch, I just moved it from `internal/cli/main.ts` into here. Definitely something to think about moving forward though.